### PR TITLE
Pokemod Fossil, Jungle, Black Star Promos

### PR DESCRIPTION
**Pull request by TheAquaPiplup**
bad

### CODE REVIEW

**Review by flagrama — CHANGES_REQUESTED** [review]
I've only reviewed the rares so far. You also don't need `PokemodFossil.` in front of the enum value in any of the copies you are doing since they are in that same file.

You also need to replace the imports at the top with this:
```
package tcgwars.logic.impl.pokemod

import tcgwars.logic.impl.gen1.Fossil

import static tcgwars.logic.card.HP.*;
import static tcgwars.logic.card.Type.*;
import static tcgwars.logic.card.CardType.*;
import static tcgwars.logic.groovy.TcgBuilders.*;
import static tcgwars.logic.groovy.TcgStatics.*
import static tcgwars.logic.effect.ability.Ability.ActivationReason.*
import static tcgwars.logic.effect.EffectType.*;
import static tcgwars.logic.effect.Source.*;
import static tcgwars.logic.effect.EffectPriority.*
import static tcgwars.logic.effect.special.SpecialConditionType.*
import static tcgwars.logic.card.Resistance.ResistanceType.*

import java.util.*;
import org.apache.commons.lang.WordUtils;
import tcgwars.entity.*;
import tcgwars.logic.*;
import tcgwars.logic.card.*;
import tcgwars.logic.card.energy.*;
import tcgwars.logic.card.pokemon.*;
import tcgwars.logic.card.trainer.*;
import tcgwars.logic.effect.*;
import tcgwars.logic.effect.ability.*;
import tcgwars.logic.effect.advanced.*;
import tcgwars.logic.effect.basic.*;
import tcgwars.logic.effect.blocking.*;
import tcgwars.logic.effect.event.*;
import tcgwars.logic.effect.getter.*;
import tcgwars.logic.effect.special.*;
import tcgwars.logic.util.*;

/**
 * @author axpendix@hotmail.com
 */
```
Otherwise this will just cause a bunch of errors.

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 1]
```diff
@@ -130,129 +130,801 @@ public enum PokemodFossil implements LogicCardInfo {
...
+            text "Once during your turn (before your attack), you may move 1 damage counter from 1 of your opponent’s Pokémon to another (even if it would Knock Out the other Pokémon). This power can’t be used if Gengar is Asleep, Confused, or Paralyzed."
+            actionA {
+              checkLastTurn()
+              assert opp.bench : "There is only one Pokémon"
+              assert opp.all.find{it.numberOfDamageCounters} : "None of your opponent’s Pokémon have damage counter to move"
+              powerUsed()
+              def src = opp.all.findAll{it.numberOfDamageCounters}.select()
+              def tar = opp.all.findAll{it != src}.select()
+              src.damage-=hp(10)
+              tar.damage+=hp(10)
```
[the remark this answers is not in the document]
Comment (starg09) [reply src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:180]: Yeah, though I'd still manually reduce the src.damage value, using directDamage for tar.

> Effects that prevent you from healing or from removing damage counters do not affect moving those damage counters between Pokémon. (Dec 1, 2011 TPCi Rules Team)

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 2]
```diff
@@ -130,129 +130,801 @@ public enum PokemodFossil implements LogicCardInfo {
...
+              def selfDmg = 0
+              if(opp.bench){
+                def benchDmg = false
+                opp.bench.each{
+                  benchDmg = false
+                  flip 1,{
+                    benchDmg= true
+                  },{selfDmg += 1}
+                  if(benchDmg) damage 20, it
+                }
```
[the remark this answers is not in the document]
Comment (starg09) [reply src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:406]: Wording is changed in Pokemod's version it seems, but Pluspower still only affects the Defending Pokémon apparently.

The sum of tails for a final self damage sounds good, less spam in the log 👍 

Would maybe add a string at the start of Flip, as to differentiate for Which Pokémon the flip is for.

[the remark this answers is not in the document]
Comment (flagrama) [reply src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:406]: Would it spam? Most of the time Damage gets added together, then applied after the closure, doesn't it?

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 3]
```diff
@@ -172,9 +248,10 @@ public enum PokemodFossil implements LogicCardInfo {
             actionA {
               checkLastTurn()
               assert opp.bench : "There is only one Pokémon"
-              assert opp.all.find{it.numberOfDamageCounters} : "None of your opponent’s Pokémon have damage counter to move"
+              def choices = opp.all.findAll {it.numberOfDamageCounters}
+              assert choices : "None of your opponent’s Pokémon have damage counter to move"
               powerUsed()
-              def src = opp.all.findAll{it.numberOfDamageCounters}.select()
+              def src =chocies.select()
```
Comment (flagrama) [inline src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:254]: You're gonna have to double check your spelling here.

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 4]
```diff
@@ -115,18 +115,19 @@ public enum PokemodFossil implements LogicCardInfo {
...
+  DRAGONITE_79 ("Dragonite", "78", Rarity.HOLORARE, [STAGE2, EVOLUTION, POKEMON, _COLORLESS_]),
+  GENGAR_80 ("Gengar", "79", Rarity.HOLORARE, [STAGE2, EVOLUTION, POKEMON, _PSYCHIC_]),
+  HO_OH_81 ("Ho-oh", "80", Rarity.RARE, [BASIC, POKEMON, _COLORLESS_]),
+  LUGIA_82 ("Lugia", "81", Rarity.RARE, [BASIC, POKEMON, _COLORLESS_]),
+  ARTICUNO_83 ("Articuno", "82", Rarity.RARE, [BASIC, POKEMON, _WATER_]),
+  MOLTRES_84 ("Moltres", "83", Rarity.RARE, [BASIC, POKEMON, _FIRE_]),
+  ZAPDOS_85 ("Zapdos", "84", Rarity.RARE, [BASIC, POKEMON, _LIGHTNING_]),
+  ARTICUNO_EX_86 ("Articuno", "85", Rarity.ULTRARARE, [BASIC, POKEMON, _WATER_]),
+  MOLTRES_EX_87 ("Moltres", "86", Rarity.ULTRARARE, [BASIC, POKEMON, _FIRE_]),
+  ZAPDOS_EX_88 ("Zapdos", "87", Rarity.ULTRARARE, [BASIC, POKEMON, _LIGHTNING_]),
```
Comment (starg09) [inline src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:130]: Remember to fix the collectionValue if changing the number in the enum :)

```suggestion
  PSYDUCK_78 ("Psyduck", "78", Rarity.RARE, [BASIC, POKEMON, _WATER_]),
  DRAGONITE_79 ("Dragonite", "79", Rarity.HOLORARE, [STAGE2, EVOLUTION, POKEMON, _COLORLESS_]),
  GENGAR_80 ("Gengar", "80", Rarity.HOLORARE, [STAGE2, EVOLUTION, POKEMON, _PSYCHIC_]),
  HO_OH_81 ("Ho-oh", "81", Rarity.RARE, [BASIC, POKEMON, _COLORLESS_]),
  LUGIA_82 ("Lugia", "82", Rarity.RARE, [BASIC, POKEMON, _COLORLESS_]),
  ARTICUNO_83 ("Articuno", "83", Rarity.RARE, [BASIC, POKEMON, _WATER_]),
  MOLTRES_84 ("Moltres", "84", Rarity.RARE, [BASIC, POKEMON, _FIRE_]),
  ZAPDOS_85 ("Zapdos", "85", Rarity.RARE, [BASIC, POKEMON, _LIGHTNING_]),
  ARTICUNO_EX_86 ("Articuno", "86", Rarity.ULTRARARE, [BASIC, POKEMON, _WATER_]),
  MOLTRES_EX_87 ("Moltres", "87", Rarity.ULTRARARE, [BASIC, POKEMON, _FIRE_]),
  ZAPDOS_EX_88 ("Zapdos", "88", Rarity.ULTRARARE, [BASIC, POKEMON, _LIGHTNING_]),
```

**Comment by flagrama** [comment]
Ah, I would request that you create new PRs for different files. Otherwise this one gets even harder and harder to review.

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 5]
```diff
@@ -130,129 +182,871 @@ public enum PokemodFossil implements LogicCardInfo {
...
+            }
+          }
+          move "Wing Attack", {
+            text "30 damage."
+            energyCost C, C, C
+            attackRequirement {}
+            onAttack {
+              damage 30
+            }
+          }
```
Comment (flagrama) [inline src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:208]: This is a pokeBody in Pokémod, not a Pokémon Power. You should also update the "text" value where appropriate to match what is on the Pokémod card. Keeping the text accurate helps with reviewing the changes to ensure they are accurate to the text on the card without having to load the carddb in the browser.

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 6]
```diff
@@ -130,129 +182,871 @@ public enum PokemodFossil implements LogicCardInfo {
...
+            text "No more Evolution cards can be played. This power stops working while Aerodactyl is Asleep, Confused, or Paralyzed."
+            delayedA {
+              if (self.cards.filterByType(POKEMON_TOOL)) {
+              before PLAY_EVOLUTION, {
+                if(!self.specialConditions){
+                  wcu "Prehistoric Power prevents you from playing Evolution cards!"
+                  prevent()
+                }
+              }
+            }
```
Comment (flagrama) [inline src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy:198]:
```suggestion
              before PLAY_EVOLUTION, {
                if(!self.specialConditions && self.cards.filterByType(POKEMON_TOOL)){
                  wcu "Prehistoric Power prevents you from playing Evolution cards!"
                  prevent()
                }
              }
            }
```
You need to check the tool is attached inside of the `before`. Checking it outside will prevent the delayed effect from ever being registered if you don't have a tool attached when Aerodactly comes into play, even if you attach it later.

File: src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy [hunk 7]
```diff
@@ -386,12 +387,12 @@ public enum PokemodFossil implements LogicCardInfo {
...
           weakness LIGHTNING
           move "Water Gun", {
-            text "10+ damage. Does 10 damage plus 10 more damage for each [W] Energy attached to Lapras but not used to pay for this attack’s Energy cost. You can’t add more than 20 damage in this way."
+            text "10+ damage. Does 10 damage plus 10 more damage for each [W] E."
             energyCost W
             attackRequirement {}
             onAttack {
               damage 10
-              extraEnergyDamage(2,hp(10),W,thisMove)
+              extraEnergyDamage(2,hp(10),W,thisMove)nergy attached to Lapras but not used to pay for this attack’s Energy cost. You can’t add more than 20 damage in this way
```
Comment (flagrama) [inline src/tcgwars/logic/impl/pokemod/PokemodFossil.groovy]: This needs fixed since it'll cause an error.

**Comment by flagrama** [comment]
GitHub doesn't seem to be showing my rebase and fix build force push in it's log here, but the files seem to have been updated. If everything seems alright after this comment, I'm just going to merge this so this can actually be tested and bugs fixed. It's way to large to review easily.

**Pull request merged**